app/src/matrix_transform.c

```diff
@@ -33,7 +36,12 @@ uint32_t zmk_matrix_transform_row_column_to_position(uint32_t row, uint32_t colu
     matrix_index = (row * ZMK_MATRIX_COLS) + column;
 
 #ifdef ZMK_KEYMAP_TRANSFORM_NODE
-    return transform[matrix_index];
+    if (matrix_index >= ARRAY_SIZE(transform))
+        return -EINVAL;
+    value = transform[matrix_index];
```

Comment:
Minor: [Zephyr requires](https://docs.zephyrproject.org/3.1.0/develop/languages/c/index.html#language-standards) a compiler that supports C99 (and some C11 features), so variables don't need to be declared at the top of a scope. Declaring and initializing a variable in the same statement prevents accidentally using it before it's initialized.

```suggestion
    const uint32_t value = transform[matrix_index];
```

(could also do the same for `matrix_index` if you wanted)

Reply:
DONE